libraries/freertos_plus/standard/freertos_plus_posix/test/iot_test_posix_pthread.c

```diff
@@ -364,6 +411,182 @@ TEST( Full_POSIX_PTHREAD, pthread_create_join )
 
 /*-----------------------------------------------------------*/
 
+TEST( Full_POSIX_PTHREAD, pthread_detach_before_running )
+{
+    int iThreadCreateStatus = 0, iStatus = 0;
+    pthread_t xNewThread = NULL;
+    pthread_barrier_t xBarrier = { 0 };
+
+    /* Initialize a pthread barrier for two threads. */
+    iStatus = pthread_barrier_init( &xBarrier, NULL, 2U );
+    TEST_ASSERT_EQUAL_INT( 0, iStatus );
+
+    if( TEST_PROTECT() )
+    {
+        /* Suspend the scheduler to prevent a context switch. */
+        vTaskSuspendAll();
+        {
+            /* Create a new thread within a critical section. By default a joinable thread is created. */
+            iThreadCreateStatus = pthread_create( &xNewThread, NULL, prvBarrierThread, ( void * ) &xBarrier );
+
+            /* Detach the thread before the new thread starts execution. */
+            if( iThreadCreateStatus == 0 )
+            {
+                iStatus = pthread_detach( xNewThread );
+            }
+        }
+        ( void ) xTaskResumeAll();
+
+        TEST_ASSERT_EQUAL_MESSAGE( 0, iThreadCreateStatus, "Thread creation failed." );
+        TEST_ASSERT_EQUAL_MESSAGE( 0, iStatus, "Thread detach failed" );
+
+        /*
+         * Wait on the barrier for new thread to start execution.
+         * Once barrier is released, new thread exits and releases resource by itself.
```

Comment:
Is there a way to verify that the thread did exit and released resources?

Reply:
Pthread implementation uses opaque pointers, so test by itself cannot access the actual FreeRTOS task handle being created.